src/modules/api.js

```diff
@@ -0,0 +1,125 @@
+import ui from './ui.js';
+import resBtn from './reservation.js';
+
+const id = 'tKVlvnEbmf4TMWB77SE7';
+const urlMealrecipe = 'https://www.themealdb.com/api/json/v1/1/lookup.php?i=';
+const urlAllMeals = 'https://www.themealdb.com/api/json/v1/1/filter.php?c=Seafood';
+const invApiUrl = `https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/${id}/likes`;
+export const ides = ['52959', '52819', '52944', '53043', '52802', '52918'];
+
+const getLikes = async () => {
+  try {
+    const allLikes = await fetch(invApiUrl, {
+      method: 'Get',
+    });
+    return allLikes.json();
+  } catch (error) {
+    return error;
+  }
+};
+
+const postReservations = async (data) => {
+  let reservation = '';
+  try {
+    reservation = await fetch(
+      'https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/tKVlvnEbmf4TMWB77SE7/reservations/',
+      {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify(data),
+      },
+    );
+    // eslint-disable-next-line no-empty
+  } catch (error) {}
+  return reservation;
+};
+
+const getReservations = async (id) => {
+  const ID = id.toString();
+  try {
+    const reservation = await fetch(
+      `https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/tKVlvnEbmf4TMWB77SE7/reservations?item_id=${ID}`,
+      {
+        method: 'Get',
+      },
+    );
+    const res = await reservation.json();
+    return res;
+  } catch (error) {
+    return error;
+  }
+};
+
+const getStrInstruction = async (id) => {
+  const ID = id.toString();
+  try {
+    const reservation = await fetch(
+      `https://www.themealdb.com/api/json/v1/1/lookup.php?i=${ID}`,
+      {
+        method: 'Get',
+      },
+    );
+    const res = await reservation.json();
+    return res.meals[0];
+  } catch (error) {
+    return error;
+  }
+};
+
+const getMealsInfo = async () => {
+  const allMeals = await fetch(urlAllMeals, {
+    method: 'Get',
+  });
+  const likes = await getLikes();
+  const meals = await allMeals.json();
+  ui(meals.meals, likes);
+  resBtn(meals.meals);
+};
+
+const addLike = async (index) => {
+  try {
+    const userPost = await fetch(invApiUrl, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        item_id: `${ides[index]}`,
+      }),
+    });
+    return userPost;
+  } catch (error) {
+    return error;
+  }
+};
```

Comment:
- [ ] OPTIONAL
Great job, _like_ feature properly implemented! However, I believe we can improve the UX of our app by adding an _unlike_ feature also, to enable a user can easily revert a _like_ action that is unintended.

Reply:
Dear @rloterh, actually we wanted to do this also in the first, but the API that [Microverse template](https://www.notion.so/Involvement-API-869e60b5ad104603aa6db59e08150270) gave us for this project doesn't have a POST request to decrease or delete likes values, we only can send the item id in request and it increases it by one unit, we appreciate any suggestion about how we can do this?

Reply:
Hi @mohisa302 

I recall couple of other students implement the _unlike_ feature, This project is fairly new and I haven't worked on it myself yet. Meanwhile let look into it and I'll get back to you. 

Cheers! 😃